db/migrate/20210128011516_create_users.rb

```diff
@@ -8,14 +8,20 @@ def change
       t.string :username, null: false
       t.index :username, unique: true
 
-      t.string :email, null: false
+      t.string :email
       t.index :email, unique: true
 
+      t.string :unconfirmed_email
+      t.index :unconfirmed_email
+
       t.string :password_digest
 
       t.string :auth_token
       t.index :auth_token, unique: true
 
+      t.string :email_confirmation_token
+      t.index :email_confirmation_token, unique: true
+
```

Comment:
Typically we'd want to create a separate migration for this type of stuff btw, but because this is a weird case editing the old one is better. But for future PRs all migrations should be new.